testing/reconciler.go

```diff
@@ -119,6 +122,30 @@ func (tc *ReconcilerTestCase) Run(t *testing.T, scheme *runtime.Scheme, factory
 	}
 
 	ctx := context.Background()
+	ctx = logr.NewContext(ctx, logrtesting.NewTestLogger(t))
+	if deadline, ok := t.Deadline(); ok {
```

Comment:
**question:** Should the go doc for each TestCase mention that the ctx is cancelable?

Reply:
Probably not. I'm not sure how to communicate the meaning short of saying "it's a context, treat it is as such".

It's not user cancelable, and all contexts passed into a method are inherently potentially cancelable. There's a lot of discussion upstream in https://github.com/golang/go/issues/28135. The main benefit is if there are any go routines still in flight after the test returns they can react to the context being canceled (we don't use parallel tests). Previously we used a Background context that would remain open forever, until the test runner stops the execution dead.

Reply:
>Probably not. I'm not sure how to communicate the meaning short of saying "it's a context, treat it is as such".

`+1`